agent/agent.go

```diff
@@ -87,14 +87,19 @@ func hostProxy(ctx context.Context, host, shimPath string, injectShimCode bool)
 	})
 	hostProxy.FlushInterval = 100 * time.Millisecond
 	var h http.Handler = hostProxy
+	h = sessionLRU.SessionHandler(h)
 	if shimPath != "" {
 		var err error
-		h, err = websockets.Proxy(ctx, hostProxy, host, shimPath, injectShimCode, *rewriteWebsocketHost)
+		h, err = websockets.Proxy(ctx, h, host, shimPath, *rewriteWebsocketHost, sessionLRU.SessionHandler)
```

Comment:
Please document the subtle interactions between the session handler and the websockets proxy that require the session handler to be effectively passed in twice.

Reply:
Done